learning_journal/tests.py

```diff
@@ -1,65 +1,249 @@
-import unittest
+import pytest
 import transaction
-
+import time
 from pyramid import testing
 
-
-def dummy_request(dbsession):
-    return testing.DummyRequest(dbsession=dbsession)
-
-
-class BaseTest(unittest.TestCase):
-    def setUp(self):
-        self.config = testing.setUp(settings={
-            'sqlalchemy.url': 'sqlite:///:memory:'
-        })
-        self.config.include('.models')
-        settings = self.config.get_settings()
-
-        from .models import (
-            get_engine,
-            get_session_factory,
-            get_tm_session,
-            )
-
-        self.engine = get_engine(settings)
-        session_factory = get_session_factory(self.engine)
-
-        self.session = get_tm_session(session_factory, transaction.manager)
-
-    def init_database(self):
-        from .models.meta import Base
-        Base.metadata.create_all(self.engine)
-
-    def tearDown(self):
-        from .models.meta import Base
-
+from learning_journal.models import (
+    Entry,
+    get_tm_session,
+)
+from learning_journal.models.meta import Base
+from learning_journal.scripts.initializedb import ENTRIES
+from pyramid.config import Configurator
+import datetime
+
+TEST_ENTRIES = [
+    {'title': "Test 1",
+     'body': 'Test body: first entry',
+     'creation_date': datetime.datetime(2016, 12, 24, 0, 0),
+     'id': 1},
+    {'title': "Test 2",
+     'body': 'Test body: second entry',
+     'creation_date': datetime.datetime(2017, 1, 1, 0, 0),
+     'id': 2},
+    {'title': "Test 3",
+     'body': 'Test body: third entry',
+     'creation_date': datetime.datetime(2017, 1, 2, 0, 0),
+     'id': 3},
+    {'title': "Test 4",
+     'body': 'Test body: fourth entry',
+     'creation_date': datetime.datetime(2017, 1, 3, 0, 0),
+     'id': 4},
+    {'title': "Test 5",
+     'body': 'Test body: fifth entry',
+     'creation_date': datetime.datetime(2017, 1, 4, 0, 0),
+     'id': 5},
+]
+
+# Unit tests
+
+@pytest.fixture(scope="session")
+def configuration(request):
+    """Set up a Configurator instance.
+
+    This Configurator instance sets up a pointer to the location of the
+        database.
+    It also includes the models from your app's model package.
+    Finally it tears everything down, including the in-memory SQLite database.
+
+    This Configurator sets up a pointer to the location of the database.
+    It also includes the models from your app's model package.
+    Finally it tears everything down, including the in-memory SQLite database.
+    This configuration will persist for the entire duration of your PyTest run.
+    """
+    config = testing.setUp(settings={
+        'sqlalchemy.url': 'postgres://hotsauce@localhost:5432/learning_journal'
+    })
+    config.include(".models")
```

Comment:
If you don't include `routes` as well, any unit test that does a redirect is guaranteed to break